server/bin/pbench-server-put-shim-001.py

```diff
@@ -0,0 +1,176 @@
+) -> Results:
+    """process_tb - receives a tar ball and copies it to the remote server
+                    using the server's HTTP PUT method
+
+        Args -
+            config -- PbenchServer config object
+            logger -- logger object to use when emitting log entries during
+                      operation
+            receive_dir -- directory where tarballs are received from agent
+
+    """
+
+    # Check for results that are ready for processing: version 002 agents
+    # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+    # after they are done with MD5 checking so that's what we look for.
+    list_check = glob.glob(
+        os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
+    )
+
+    logger.info("{}", config.TS)
+    list_check.sort()
+    nstatus = ""
+
+    ntotal = ntbs = 0
+
+    # FIXME: keep a generated token which has longer duration
+
+    token = config.get("pbench-server", "put-token")
+
+    for tbmd5 in list_check:
+        ntotal += 1
+
+        # full pathname of tarball
+        tb = Path(tbmd5[0:-4])
+        tbmd5 = Path(tbmd5)
+        tbdir = tb.parent
+        controller = tbdir.name
+
+        try:
+            tarball_len, tarball_md5 = md5sum(tb)
+            crt = CopyResultTb(
+                controller, tb, tarball_len, tarball_md5, config, logger,
+            )
+            crt.copy_result_tb(token)
+        except Exception as e:
+            logger.error("Unexpected Error: {}", e)
+            continue
+
+        try:
+            os.remove(tbmd5)
+        except Exception as exc:
+            logger.error(
+                "{}: Warning: cleanup of successful copy operation failed: '{}'",
+                config.TS,
+                exc,
+            )
+
+        try:
+            os.remove(tb)
+        except Exception as exc:
+            logger.error(
+                "{}: Warning: cleanup of successful copy operation failed: '{}'",
+                config.TS,
+                exc,
+            )
+
+        ntbs += 1
+
+        nstatus = f"{nstatus}{config.TS}: processed {tb}\n"
+        logger.info(f"{tb.name}: OK")
+
+    return Results(nstatus, ntotal, ntbs)
+
+
+def main(cfg_name: PbenchServerConfig) -> int:
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable or use --config <file> on the"
+            " command line",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchServerConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e} (config file {cfg_name})", file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    receive_dir = get_receive_dir(config, logger)
+
+    if receive_dir is None:
+        return 2
+
+    counts = process_tb(config, logger, receive_dir)
+
+    # FIXME: I believe we can just log the status instead of creating report
+
```

Comment:
I believe that you have changed your mind about this.  ;-)

I don't have an opinion about whether we just log the status or index a report, but I am generally against merging code with `FIXME`'s in it.  :-)

Reply:
I will remove the FIXME :)